parcel-bundler/parcel
# Break up request graph cache serialisation and run after build completion

# ↪️ Pull Request

Currently in Jira Frontend, we have a major problem with developers becoming confused when shutting down parcel, as it looks like the process has crashed. This is because the request tracker is serialised and written to cache, when the parcel cli is shutdown.

To avoid this, we can utilise the idle time in the watch process by serialising after the build has completed. It's also important to break up the serialisation phase, as we need to be able to interrupt at any time for a new build. I picked a sane constant value to split up the nodes in groups that took around ~10ms to process on my machine, which should free up the event loop well enough for most developers.

## 🚨 Test instructions

1. Start a parcel build in watch mode
2. Wait for build to complete
3. Observe cache serialising to disk
4. Close parcel (observe near-instant shutdown)
